### PR TITLE
feat: header bell icon

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- The primary goal in this PR is to add the bell icon on the header.
- It can be noticed that there seem to be a lot of changes that happened in this PR. I decided to refactor the `MainLayout` to extract all the `Header` related concerns. Everything related to it can be found in this commit https://github.com/dailydotdev/apps/commit/2144bab87e8febdfd150b4b047774a85946e0785
- To make it easier to review the changes for applying the new bell icon, please refer to this specific commit https://github.com/dailydotdev/apps/commit/93f7d402ce7fcdd10dedea5c93a922c64848418f

Note: as mentioned in the primary PR, the number indication shall be worked on once the integration has been created.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-695 #done

### CODE REVIEW

File: packages/shared/src/components/layout/MainLayoutHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            )}
+            {shouldShowLogo({ mobileTitle, sidebarRendered }) && (
+              <HeaderLogo
+                user={user}
+                onLogoClick={onLogoClick}
+                greeting={greeting}
+              />
+            )}
+          </div>
+          {!hideButton && user && (
```
Comment (rebelchris) [inline packages/shared/src/components/layout/MainLayoutHeader.tsx:91]: When do we want this one hidden?

**Review by rebelchris — APPROVED** [review]
Wow thanks for abstracting so much here 🙏

File: packages/shared/src/components/layout/MainLayoutHeader.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/layout/MainLayoutHeader.tsx:91]: When things are still being loaded and when we only show the logo parameter is true.

File: packages/shared/src/components/layout/MainLayoutHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            <LinkWithTooltip
+              tooltip={{ placement: 'left', content: 'Notifications' }}
+              href={`${webappUrl}notifications`}
```
Comment (el) [inline packages/shared/src/components/layout/MainLayoutHeader.tsx:94]: what happens to this link if it is used as an extension?

**Review by el — APPROVED** [review]
Just had a small question, other than that LGTM. Thanks ❤️

File: packages/shared/src/components/layout/MainLayoutHeader.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/layout/MainLayoutHeader.tsx:94]: The extension will redirect you to our webapp. A similar scenario would be from the extension and then clicking the bookmarks tab.

**Pull request merged**